velox/dwio/parquet/tests/ParquetTpchTestBase.h

```diff
@@ -86,22 +89,77 @@ class ParquetTpchTestBase : public testing::Test {
   }
 
  private:
+  void initializeTpch(const double scaleFactor) {
+    auto query = fmt::format("CALL dbgen(sf={})", scaleFactor);
+    duckDb_->execute(query);
+  }
+
+  std::string getTpchQuery(int queryNo) const {
+    auto queryString = ::duckdb::TPCHExtension::GetQuery(queryNo);
+    // Output of GetQuery() has a new line and a semicolon. These need to be
+    // removed in order to use the query string in a subquery
+    queryString.pop_back(); // remove new line
+    queryString.pop_back(); // remove semi-colon
+    return queryString;
+  }
+
   /// Write TPC-H tables as a Parquet file to temp directory in hive-style
   /// partition
-  void saveTpchTablesAsParquet() {
+  void saveTpchTablesAsParquet(
+      std::shared_ptr<connector::Connector> hiveConnector) {
     constexpr int kRowGroupSize = 10'000;
     const auto tableNames = tpchBuilder_.getTableNames();
     for (const auto& tableName : tableNames) {
       auto tableDirectory =
           fmt::format("{}/{}", tempDirectory_->path, tableName);
-      fs::create_directory(tableDirectory);
-      auto filePath = fmt::format("{}/file.parquet", tableDirectory);
-      auto query = fmt::format(
-          fmt::runtime(kDuckDbParquetWriteSQL_.at(tableName)),
-          tableName,
-          filePath,
-          kRowGroupSize);
-      duckDb_->execute(query);
+      auto query =
+          fmt::format(fmt::runtime(kDuckDbTpchSQL_.at(tableName)), tableName);
+
+      auto res = duckDb_->execute(query);
+
+      auto schema = res->getType();
+
+      auto locationHandle = std::make_shared<connector::hive::LocationHandle>(
+          tableDirectory,
+          tableDirectory,
+          connector::hive::LocationHandle::TableType::kNew);
+      std::vector<std::shared_ptr<const connector::hive::HiveColumnHandle>>
+          inputColumns;
+
+      for (auto tpe : schema->children()) {
+        inputColumns.push_back(
+            std::make_shared<connector::hive::HiveColumnHandle>(
+                tpe->name(),
+                connector::hive::HiveColumnHandle::ColumnType::kRegular,
+                tpe));
+      }
+
+      auto config = std::make_unique<core::MemConfig>();
```

Comment:
Do the config and conQueryCtx need to be created per table ? Once for all the tables should suffice I think.